src/main.py

```
@@ -0,0 +1,325 @@
+        # Check if directory exists, if not create it with first id = 1
+        if not os.path.exists(data_dir):
+            os.makedirs(data_dir)
+            next_id = 1
+        else:
+            # List containing all existing ids
+            existing_ids = []
+            # Loop through all files in directory with their names ['1.json', '2.json', etc...]
+            for filename in os.listdir(data_dir):
+                # Check if file ends with .json
+                if filename.endswith('.json'):
+                    # Convert filename to integer
+                    file_id = int(filename.replace('.json', ''))
+                    # Add found id to list of existing ids
+                    existing_ids.append(file_id)
+            # If list of existing ids is not empty, find highest id and increment by 1
+            if existing_ids:
+                next_id = max(existing_ids) + 1
+            # If list of existing ids is empty, first id will be 1
+            else:
+                next_id = 1
+        # Add the generated ID to the request data
+        data['id_user'] = next_id
+        # Save user data to JSON file
+        filename = os.path.join(os.getcwd(), 'data', 'users', f'{next_id}.json')
+        with open(filename, 'w', encoding='utf-8') as f:
+            json.dump(data, f, indent=4, ensure_ascii=False)
+        return jsonify({'message': 'Usuário criado com sucesso!',
+                        'id_user': data['id_user']}), 200
+    except Exception as e:
+        return jsonify({'message': 'Erro interno do servidor.',
+                        'error': str(e)}), 500
+
+@app.route('/users', methods=['GET'])
+def get_users():
+    """Return a list of all users from the data directory."""
+    try:
+        users = []
+        data_dir = os.path.join(os.getcwd(), 'data', 'users')
+        # Read all JSON files in the data directory
+        for filename in os.listdir(data_dir):
```

One more think about new version, maybe you can return a error when you can't find the user "User not found"

---

Done! Added additional error handling when reading user and company files.